File: examples/java/echo/src/main/java/ftl/echo/Echo.java
package ftl.echo;

import ftl.time.TimeClient;
import xyz.block.ftl.Export;
import xyz.block.ftl.Verb;

public class Echo {

    @Export
    @Verb
    public EchoResponse echo(EchoRequest req, TimeClient time) {
        var response = time.time();
        return new EchoResponse("Hello, " + req.name().orElse("anonymous") + "! The time is " + response.toString() + ".");
    }
}

File: examples/kotlin/echo/src/main/kotlin/ftl/echo/Echo.kt
package ftl.echo

import ftl.time.TimeClient
import xyz.block.ftl.Export
import xyz.block.ftl.Verb

data class EchoRequest(val name: String?)
data class EchoResponse(val message: String)

@Export
@Verb
fun echo(req: EchoRequest, time: TimeClient): EchoResponse {
  val response = time.call()
  return EchoResponse(message = "Hello, ${req.name ?: "anonymous"}! The time is ${response.time}.")
}
